util/stream_dataset.py

```diff
@@ -409,7 +388,7 @@ def _insert_data(dataset, param, data, provenance_metadata=None, request_id=None
             # remove obs dimension from parameter's dimensions and data (13025 AC2)
             param_dimensions.remove('-obs')
             dims = param_dimensions
-            data = data[0] if data else None
+            data = data[0] if data is not None else None
```

Comment:
While this better fulfills the comments, my understanding is it will work only as long as data is populated and meets the sequence protocol (unless it is None). It appears this is dependent upon what's returned in "result" from the calls made in _execute_algorithm(). Is this a reasonable assumption?

Reply:
In the _try_create_derived_product() method, the "result" variable is set to the first part of the return value of _execute_algorithm(). I agree that at that point the type of "result" is in question, although it should be a numpy.ndarray. That being said, immediately after the _execute_algorithm() call, a check is made to see if the "result" type is numpy.ndarray, and if not, "results" is caste to a numpy.ndarray. Should the caste fail, I would expect exception handling to ensue and _insert_data() not to be called. So, if the _insert_data() is called from the try block of _try_create_derived_product(), it is called with the "data" input being of type numpy.ndarray. The other two calls I see to _insert_data() both have a "data" value of None.

Reply:
The original issue here was that the if data check does not work the same way on numpy.ndarrays as it does on lists, sets, etc.

>>> a = numpy.ndarray([1, 2, 3])
>>> a[0] if a else None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()